[debug] > Exec(all {file:/C:/repo/scala3_continuation/}root/products {file:/C:/repo/scala3_continuation/}root/test:products, None, None)
[debug] Evaluating tasks: Compile / products, Test / products
[debug] Running task... Cancel: Signal, check cycles: false, forcegc: true
[info] compiling 2 Scala sources to C:\repo\scala3_continuation\target\scala-3.3.4\classes ...
[error] -- [E007] Type Mismatch Error: C:\repo\scala3_continuation\src\main\scala\CpsAsyncTest.scala:12:57 
[error] 12 |      await(cont(await(Future.successful(()).asInstanceOf[T])))
[error]    |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[error]    |Found:    T
[error]    |Required: F[T²]
[error]    |
[error]    |where:    F  is a type variable with constraint <: [_] =>> Any
[error]    |          T  is a type in method shift
[error]    |          T² is a type variable with constraint <: T
[error]    |
[error]    |Note that implicit conversions were not tried because the result of an implicit conversion
[error]    |must be more specific than F[T]
[error]    |
[error]    | longer explanation available when compiling with `-explain`
[error] -- [E172] Type Error: C:\repo\scala3_continuation\src\main\scala\CpsAsyncTest.scala:12:61 
[error] 12 |      await(cont(await(Future.successful(()).asInstanceOf[T])))
[error]    |                                                             ^
[error]    |No given instance of type cps.CpsAwaitable[[_] =>> Any] was found for parameter am of method await in package cps
[error] -- [E006] Not Found Error: C:\repo\scala3_continuation\src\main\scala\CpsAsyncTest.scala:24:25 
[error] 24 |    val result = shift { cont: Unit =>
[error]    |                         ^^^^
[error]    |  Not found: cont - did you mean Long? or perhaps Long or None or clone?
[error]    |
[error]    | longer explanation available when compiling with `-explain`
[error] three errors found
